src/speech_to_text_finetune/data_process.py

```diff
@@ -97,87 +122,188 @@ def load_dataset_from_dataset_id(
+
+        dev_df = pd.read_csv(cv_data_dir / "dev.tsv", sep="\t")
+        dev_df["split"] = "dev"
+
+        test_df = pd.read_csv(cv_data_dir / "test.tsv", sep="\t")
+        test_df["split"] = "test"
+
+        dataset_df = pd.concat([train_df, dev_df, test_df], ignore_index=True)
+        audio_dir = cv_data_dir / "clips"
+        audio_clip_column = "path"
+
+    return _build_cv_dataset_from_df(
+        dataset_df=dataset_df,
+        audio_dir=audio_dir,
+        audio_clip_column=audio_clip_column,
+        is_spontaneous_speech=is_spontaneous_speech,
+    )
 
-    Returns:
-        DatasetDict: HF Dataset dictionary that consists of two distinct Datasets (train+validation and test)
+
+def _build_cv_dataset_from_df(
+    dataset_df: pd.DataFrame,
+    audio_dir: str | Path,
+    audio_clip_column: str,
+    is_spontaneous_speech: bool,
+) -> DatasetDict:
     """
-    cv_data_dir = Path(cv_data_dir)
-    train_df = pd.read_csv(cv_data_dir / "train.tsv", sep="\t")
-    test_df = pd.read_csv(cv_data_dir / "test.tsv", sep="\t")
+    Common builder for Common Voice Spontaneous and Scripted Speech, MDC or local.
+    - Renames text column to `sentence` (handles SPS's column name `transcription`)
+    - Converts audio paths to absolute
+    - Merges train+dev and keeps test
+    - Returns DatasetDict with only `audio` and `sentence`
+    """
+    df = dataset_df.copy()
+
+    # Normalize text column name
+    if is_spontaneous_speech and "transcription" in df.columns:
+        df = df.rename(columns={"transcription": "sentence"})
+
+    # Ensure we have splits
+    if "split" not in df.columns:
+        raise ValueError("Expected a 'splits' column in the dataset DataFrame.")
 
-    # Replace relative path with absolute
-    train_df = train_df.rename(columns={"path": "audio"})
-    train_df["audio"] = train_df["audio"].apply(
-        lambda p: str(cv_data_dir / "clips" / p)
+    # Convert relative to absolute audio paths
+    df = _replace_rel_path_with_abs_path(
+        df=df, audio_dir=audio_dir, audio_clip_column=audio_clip_column
     )
 
-    test_df = test_df.rename(columns={"path": "audio"})
-    test_df["audio"] = test_df["audio"].apply(lambda p: str(cv_data_dir / "clips" / p))
+    # Split and keep only relevant columns
+    train_df = df[df["split"].isin(["train", "dev"])][["audio", "sentence"]]
+    test_df = df[df["split"] == "test"][["audio", "sentence"]]
 
     return DatasetDict(
         {
-            "train": Dataset.from_pandas(train_df),
-            "test": Dataset.from_pandas(test_df),
+            "train": Dataset.from_pandas(train_df, preserve_index=False),
+            "test": Dataset.from_pandas(test_df, preserve_index=False),
         }
     )
 
+def _join_audio_path(audio_dir: Path, rel_path: str) -> str:
+    """
+    Safely join base audio directory with the relative file path.
+    Keeps absolute paths as-is and resolves the result.
+    """
+    p = str(rel_path)
+    if os.path.isabs(p):
+        return p
```

Comment:
nitpick: if we're already casting rel_path, might as well make it a pathlib.Path and use `.is_absolute()` to avoid mixing os.path and pathlib.Path methods? this is a completely insignificant issue so feel free to take or leave my comment :laughing: